packages/react/src/components/Campaign/Proposition.js

```diff
@@ -51,40 +61,59 @@ const ToggleButton = styled.div`
     transform: translateY(-100%);
     background: linear-gradient(to top, rgba(255, 255, 255, 1), rgba(255, 255, 255, 0) 50%);
   }
-`;
-
-function updateHeight(comp) {
-  const campaignFormContainer = window.document.getElementById('campaign-form-container');
-  const campaignForm = window.document.getElementById('campaign-form');
-  if (campaignForm) {
-    comp.setState({ height: campaignFormContainer.offsetHeight });
-  } else {
-    setTimeout(() => updateHeight(comp), 100);
+  @media screen and (max-width: 1000px) {
+    display: ${({ containerInnerHeight }) => (containerInnerHeight > 600 ? 'block' : 'none')};
   }
-}
+`;
 
-export default class Proposition extends Component {
-  state = {
-    height: '',
-    show: false,
+export default ({
+  campaignContainerRef,
+  formContainerRef,
+  campaign,
+  children,
+  descriptionCollapse,
+  descriptionExpand,
+}) => {
+  const containerInnerRef = useRef(null);
+  const [height, setHeight] = useState(0);
+  const [show, setShow] = useState(false);
+  const [showToggle, setShowToggle] = useState(true);
+  const containerInnerHeight =
+    containerInnerRef && containerInnerRef.current && containerInnerRef.current.offsetHeight;
+  const toggleShow = () => {
+    if (show) {
+      campaignContainerRef.current.scrollIntoView({ behavior: 'smooth', block: 'start' });
+      setShow(false);
+    } else {
+      setShow(true);
+    }
   };
 
-  componentDidMount() {
-    updateHeight(this);
-  }
+  const updateHeight = () => {
+    const { offsetHeight } = formContainerRef.current;
+    setHeight(offsetHeight);
+    if (formContainerRef.current.offsetHeight > containerInnerRef.current.offsetHeight) {
+      setShowToggle(false);
+    } else {
+      setShowToggle(true);
+    }
+  };
 
-  toggleProposition = () => this.setState(({ show }) => ({ show: !show }));
+  useEffect(() => {
+    window.addEventListener('resize', updateHeight);
```

Comment:
return a cleanup func here to remove the listener as well

Reply:
@busstram done :)